File: backend/schemas/graph.py
from pydantic import BaseModel, Field
from typing import List, Optional, Any

class GraphNode(BaseModel):
    id: str = Field(..., description="unique identifier for the concept (lowercase, singular)")
    aliases: List[str] = Field(default_factory=list, description="list of alternative names or synonyms found in text")
    outbound_links: List[str] = Field(default_factory=list, description="list of concepts this node points to")
    inbound_links: List[str] = Field(default_factory=list, description="list of concepts pointing to this node")

class GraphData(BaseModel):
    nodes: List[GraphNode] = []

File: frontend-v2/src/types/graph.ts
export interface Node {
  id: string;
  aliases: string[];
  outbound_links: string[];
  inbound_links: string[];
  description?: string;
  metadata?: Record<string, any>;
}

export interface GraphData {
  nodes: Node[];
}

export interface ForceGraphNode extends Node {
  x?: number;
  y?: number;
  z?: number;
}

export interface ForceGraphLink {
  source: string;
  target: string;
}
